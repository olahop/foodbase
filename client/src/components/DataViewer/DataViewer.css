.modal {
    background-color:rgb(0,0,0,0.7);
    width: 90%;
    margin: auto
}

.modalObject{
    display: flex;
    justify-content: center;
}

.graph-button {
    border-radius: 5px;
    border: #00B4CC;
    height: 38px;
    width: 150px;
    background-color: #00B4CC;
    color: white;
    font-family: sans-serif;
    font-size: 15px;
    cursor: pointer;
}

.fa-lg {
    padding: 0 0 0 10px;
}

.no-graphics-text {
    font-size: 3em;
    background-color: white;
    padding: 5px;
}

@media (max-width: 600px) {
    .graph-button {
        font-size: 0;
        width: 50px;
        position: relative;
    }
    .fa-lg {
        font-size: initial;
    }
}
