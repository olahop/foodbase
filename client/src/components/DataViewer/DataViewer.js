import React from "react";
import Modal from "react-modal";
import Graphics from "./Graphics/Graphics";
import { connect } from "react-redux";
import "./DataViewer.css";


export class DataViewer extends React.Component {
    constructor() {
        super();

        this.state = {
            data: [],               // holds the tabledata from the redux store
            modalIsOpen: false,     // if the modal is open
            isMobile: false         // if the current screen size is typical for mobiles
        }
    
        this.openModal = this.openModal.bind(this);
        this.closeModal = this.closeModal.bind(this);
    }

    // multiple methods to manipulate isMobile state
    componentDidMount() {
        this.handleWindowSizeChange();
    }
    componentWillMount() {
        window.addEventListener("resize", this.handleWindowSizeChange);
    }
    componentWillUnmount() {
        window.removeEventListener("resize", this.handleWindowSizeChange);
    }
    handleWindowSizeChange = () => {
        if(window.innerWidth <= 500) {
          this.setState({ isMobile: true});
        } else {
          this.setState({ isMobile: false });
        }
    };

    // opens the modal
    openModal() {
        this.setState({
            modalIsOpen: true,
            data: this.props.data
        });
    }
     
    // closes the modal
    closeModal() {
        this.setState({ modalIsOpen: false});
    }

    render() {
        return (
            <>
                <div className="modalObject">
                    <button className="graph-button" onClick={this.openModal}>
                        Graphics
                        <i className="fas fa-chart-line fa-lg"></i>
                    </button>
                    <Modal
                        className="modal"
                        isOpen={this.state.modalIsOpen}
                        onAfterOpen={this.afterOpenModal}
                        onRequestClose={this.closeModal}
                        contentLabel="Graphics Modal"
                    >
                        { this.state.isMobile ? 
                        <p className="no-graphics-text">Your current screen size do not support graphics section</p>
                        :
                        <Graphics data={this.state.data}/>
                        }
                    </Modal>
                </div>
            </>
        );
    }
}

const mapStateToProps = state =>({
    data: state.data
  });

export default connect(mapStateToProps)(DataViewer)
