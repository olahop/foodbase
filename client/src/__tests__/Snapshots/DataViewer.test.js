import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import DataViewer from '../../components/DataViewer/DataViewer';

const mockStore = configureStore([]);
describe('Test App component', () => {
  let store;
  let component;
  
  beforeAll(() => {
    store = mockStore();
    component = renderer.create(
      <Provider store={store}>
        <DataViewer />
      </Provider>
    );
  });

  it('Component renders correctly', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});