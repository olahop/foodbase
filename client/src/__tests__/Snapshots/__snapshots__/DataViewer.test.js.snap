// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App component Component renders correctly 1`] = `
<div
  className="modalObject"
>
  <button
    className="graphButton"
    onClick={[Function]}
  >
    Graphics
    <i
      className="fas fa-chart-line fa-lg"
    />
  </button>
</div>
`;
